snapcraft/_options.py

```diff
@@ -280,6 +281,30 @@ def _set_machine(self, target_deb_arch):
         self.__machine_info = _ARCH_TRANSLATIONS[self.__target_machine]
 
 
+class ProjectInfo:
+    """Information gained from the snap's snapcraft.yaml file."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.name = data['name']
+        self.version = data['version']
+        self.summary = data.get('summary')
+        self.description = data.get('description')
+        self.confinement = data['confinement']
+
+
+class Project(ProjectOptions):
+    """All details around building a project concerning the build environment
+    and the snap being built."""
+
+    # Mirror ProjectOptions constructor here to guarantee order of arguments
+    # which would not be preserved with **kwargs
```

Comment:
But the only one we care about is [this](https://github.com/snapcore/snapcraft/pull/1995/files#diff-1b5470a9f2e177fc1606479628098324R68) one, right? Take the time to design this `__init__` the way we want it, and change the callers as necessary. At least make these keyword-only args, which should continue to work fine with [this call](https://github.com/snapcore/snapcraft/pull/1995/files#diff-1b5470a9f2e177fc1606479628098324R68).

Reply:
For the record: clarified in an offline conversation that this would be confusing without changing all the classes used by `Config` and porting all the consumers would make this PR too big so I'm going to do that in a follow-up branch.

Reply:
Ack. I changed the constructor and the caller in CLI.